src/zjs_fs.c

```diff
@@ -0,0 +1,1043 @@
+// Copyright (c) 2016, Intel Corporation.
+#ifdef BUILD_MODULE_FS
+
+#include <zephyr.h>
+#include <fs.h>
+
+#include <string.h>
+
+#include "zjs_callbacks.h"
+#include "zjs_common.h"
+#include "zjs_util.h"
+#include "zjs_buffer.h"
+
+#define FLAGS_READABLE      (1 << 0)
+#define FLAGS_WRITEABLE     (1 << 1)
+#define FLAGS_APPENDABLE    (1 << 2)
+#define FLAGS_CREATE        (1 << 3)
+#define FLAGS_NO_CREATE     (1 << 4)
+
+#define MODE_R          (FLAGS_READABLE | FLAGS_NO_CREATE)
+#define MODE_R_PLUS     (FLAGS_READABLE | FLAGS_WRITEABLE | FLAGS_NO_CREATE)
+#define MODE_W          (FLAGS_WRITEABLE | FLAGS_CREATE)
+#define MODE_WX         (FLAGS_WRITEABLE | FLAGS_NO_CREATE)
+#define MODE_W_PLUS     (FLAGS_WRITEABLE | FLAGS_READABLE | FLAGS_CREATE)
+#define MODE_WX_PLUS    (FLAGS_WRITEABLE | FLAGS_READABLE | FLAGS_NO_CREATE)
+#define MODE_A          (FLAGS_WRITEABLE | FLAGS_APPENDABLE | FLAGS_CREATE)
+#define MODE_AX         (FLAGS_WRITEABLE | FLAGS_APPENDABLE | FLAGS_NO_CREATE)
+#define MODE_A_PLUS     (FLAGS_READABLE | FLAGS_APPENDABLE | FLAGS_CREATE)
+#define MODE_AX_PLUS    (FLAGS_READABLE | FLAGS_APPENDABLE | FLAGS_NO_CREATE)
+
+#define MAX_PATH_LENGTH 128
+
+typedef struct file_handle_t {
+    fs_file_t fp;
+    uint16_t mode;
+    jerry_value_t fd_val;
+    int error;
+} file_handle;
```

Comment:
file_handle and file_handle_t are backwards here. The idea is file_handle_t is a newly defined type that replaces "struct file_handle".